Nanoleaf-wpf/Helpers/ScreenGrabber.cs

```diff
@@ -1,137 +1,64 @@
-﻿using NLog;
-using System;
+﻿using System;
 using System.Drawing;
 using System.Drawing.Imaging;
-using System.Runtime.InteropServices;
-using System.Timers;
-using System.Windows.Forms;
-using Winleafs.Models.Models;
 
 namespace Winleafs.Wpf.Helpers
 {
-    public static class ScreenGrabber
+    public class ScreenGrabber
     {
-        private static Logger _logger = LogManager.GetCurrentClassLogger();
-        private static object _lockObject = new object();
-
-        private static System.Timers.Timer _timer;
-        private static Rectangle _screenBounds;
-
-        private static Color _color;
-
-#pragma warning disable S3963 // "static" fields should be initialized inline
-        static ScreenGrabber()
-        {
-            var timerRefreshRate = 1000;
-
-            if (UserSettings.Settings.AmbilightRefreshRatePerSecond > 0 && UserSettings.Settings.AmbilightRefreshRatePerSecond <= 10)
-            {
-                timerRefreshRate = 1000 / UserSettings.Settings.AmbilightRefreshRatePerSecond;
-            }
-
-            _timer = new System.Timers.Timer(timerRefreshRate);
-            _timer.Elapsed += OnTimedEvent;
-            _timer.AutoReset = true;
-
-            var monitorInfo = new MonitorInfo();
-            EnumDisplaySettings(Screen.AllScreens[UserSettings.Settings.AmbilightMonitorIndex].DeviceName, -1, ref monitorInfo);
-
-            _screenBounds = new Rectangle(monitorInfo.dmPositionX, monitorInfo.dmPositionY, monitorInfo.dmPelsWidth, monitorInfo.dmPelsHeight);
-        }
-#pragma warning restore S3963 // "static" fields should be initialized inline
-
-        private static Bitmap CaptureScreen()
+        public static Bitmap CaptureScreen(Rectangle screenBounds)
```

Comment:
General idea. What if I set my monitor 2 as my screen mirror and remove the screen from my computer afterwards. Will that cause a crash?

Reply:
Good one, added checks in both the options view and the effects to catch this rare behaviour.